Add UI instructions to the Istio Access Logs tutorial (#691)
kyma-project/istio: docs/user/operation-guides/02-30-enable-istio-access-logs.md

---

You can enable [Istio access logs](https://istio.io/latest/docs/tasks/observability/logs/access-log/) to provide fine-grained details about the access to workloads that are part of the Istio service mesh. This can help indicate the four “golden signals” of monitoring (latency, traffic, errors, and saturation) and troubleshooting anomalies.
The Istio setup shipped with the Istio module provides a pre-configured [extension provider](https://istio.io/latest/docs/tasks/observability/telemetry) for access logs, which configures the Istio proxies to print access logs to stdout using the JSON format. It uses a configuration similar to the following one:

> ```bash

Reviewer: The format of this is structure and the [file](https://github.com/kyma-project/istio/blob/494f07ad9fe487ca65c5ee83f464d8a83e1c885b/internal/istiooperator/istio-operator.yaml#L153) where this configured is yaml. Why do we change it to bash?

## Configuration

Use the Telemetry API to selectively enable Istio access logs. See:
- [Configure Istio Access Logs for the Entire Namespace](#configure-istio-access-logs-for-the-entire-namespace)
- [Configure Istio Access Logs for a Selective Workload](#configure-istio-access-logs-for-a-selective-workload)
- [Configure Istio Access Logs for a Specific Gateway](#configure-istio-access-logs-for-a-specific-gateway)
- [Configure Istio Access Logs for the Entire Mesh](#configure-istio-access-logs-for-the-entire-mesh)

### Configure Istio Access Logs for the Entire Namespace

<!-- tabs:start -->

#### **Kyma Dashboard**

1. Go to the namespace for which you want to configure Istio access logs.
2. Go to **Configuration > Custom Resources**.

> 3. Scroll down to find the section `telemetry.istio.io` and select **Telemetries**.

Reviewer 2: ```suggestion
3. In the section `telemetry.istio.io`, select **Telemetries**.
```
I would avoid mentioning where exactly it is found (somewhere low, visible only after scrolling); otherwise you'll have to update the instructions if the UI changes so that scrolling down is no longer needed.

        - providers:
          - name: stdout-json
    ```
6. Replace `{YOUR_NAMESPACE}` with the name of the namespace.
7. Select **Create**.

#### **kubectl**
1. Export the name of the namespace for which you want to configure Istio access logs.
    
    ```bash
    export YOUR_NAMESPACE={NAMESPACE_NAME}
    ```

2. To apply the configuration, run:

    ```bash
    cat <<EOF | kubectl apply -f -
    apiVersion: telemetry.istio.io/v1alpha1
    kind: Telemetry
    metadata:
      name: access-config
      namespace: $YOUR_NAMESPACE
    spec:
      accessLogging:
        - providers:
          - name: stdout-json
    EOF
    ```
3. To verify that the resource is applied, run:
    ```bash
    kubectl -n $YOUR_NAMESPACE get telemetries.telemetry.istio.io
    ```
<!-- tabs:end -->

### Configure Istio Access Logs for a Selective Workload

To configure label-based selection of workloads, use a [selector](https://istio.io/latest/docs/reference/config/type/workload-selector/#WorkloadSelector).

<!-- tabs:start -->
#### **Kyma Dashboard**
1. Go to the namespace of the workloads for which you want to configure Istio access logs.
2. Go to **Configuration > Custom Resources**.

> 3. Scroll down to find the section `telemetry.istio.io` and select **Telemetries > Create Telemetry**.

Reviewer 2: same here, maybe a little more vague is still useful and easier to maintain.

```suggestion
3. In the section `telemetry.istio.io`, select **Telemetries > Create Telemetry**.
```

7. Select **Create**.

#### **kubectl**
1. Export the name of the workloads' namespace and their label as environment variables:
    
    ```bash
    export YOUR_NAMESPACE={NAMESPACE_NAME}
    export YOUR_LABEL={LABEL}
    ```

2. To apply the configuration, run:
    ```bash
    cat <<EOF | kubectl apply -f -
    apiVersion: telemetry.istio.io/v1alpha1
    kind: Telemetry
    metadata:
      name: access-config
      namespace: $YOUR_NAMESPACE
    spec:
      selector:
        matchLabels:
          service.istio.io/canonical-name: $YOUR_LABEL
      accessLogging:
        - providers:
          - name: stdout-json
    EOF
    ```
3. To verify that the resource is applied, run:
    ```bash
    kubectl -n $YOUR_NAMESPACE get telemetries.telemetry.istio.io
    ```
<!-- tabs:end -->

### Configure Istio Access Logs for a Specific Gateway

Instead of enabling the access logs for all the individual proxies of the workloads you have, you can enable the logs for the proxy used by the related Istio Ingress Gateway.

<!-- tabs:start -->
#### **Kyma Dashboard**
1. Go to the `istio-system` namespace.
2. Go to **Configuration > Custom Resources**.

> 3. Scroll down to find the section `telemetry.istio.io` and select **Telemetries**.

Reviewer 2: ```suggestion
3. In the section `telemetry.istio.io`, select **Telemetries**.
```

        - providers:
          - name: stdout-json
    ```
5. Select **Create**.

#### **kubectl**

1. To apply the configuration, run:
    ```bash
    cat <<EOF | kubectl apply -f -
    apiVersion: telemetry.istio.io/v1alpha1
    kind: Telemetry
    metadata:
      name: access-config
      namespace: istio-system
    spec:
      selector:
        matchLabels:
          istio: ingressgateway
      accessLogging:
        - providers:
          - name: stdout-json
    EOF
    ```
2. To verify that the resource is applied, run:
    ```bash
    kubectl -n istio-system get telemetries.telemetry.istio.io
    ```
<!-- tabs:end -->

### Configure Istio Access Logs for the Entire Mesh

Enable access logs for all individual proxies of the workloads and Istio Ingress Gateways.

<!-- tabs:start -->
#### **Kyma Dashboard**
1. Go to the `istio-system` namespace.
2. Go to **Configuration > Custom Resources**.

> 3. Scroll down to find the section `telemetry.istio.io` and select **Telemetries**.

Reviewer 2: ```suggestion
3. In the section `telemetry.istio.io`, select **Telemetries**.
```
